### PR TITLE
developing-layout

### PR DESCRIPTION
**Pull request by microna**

### CODE REVIEW

File: front/src/components/audit/PieChartWithPaddingAngle.jsx [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
 
-const color = "red";
+  let statusState;
+  if (componentData >= 100) {
+    statusState = "Good";
+  } else if (componentData >= 50) {
+    statusState = "Normal";
+  } else {
+    statusState = "Not Good";
+  }
```
Comment (konantovich) [inline front/src/components/audit/PieChartWithPaddingAngle.jsx]: I think it's better to do this in a separate function
Something like:

`import React from "react";
import { PieChart, Pie, Cell } from "recharts";

const checkStatusState = (componentData) => {
  if (componentData >= 100) {
   return {text: "Good", color: "#DE8344"};
  } else if (componentData >= 50) {
    return {text: "Normal", color: "#4599CF"};
  } else {
    return  {text: "Not Good", color: "#F5C342"}
  }
}

export const PieChartWithPaddingAngle = ({ componentData, status }) => {
  const data = [
    { name: "Group A", value: componentData },
    { name: "Group B", value: 100 - componentData },
  ];

  const statusState = checkStatusState(componentData)

  const COLORS = [statusState.color, "#f1f1f1"];

  return (
    <div style={{ height: "100%", width: "100%", position: "relative" }}>
      <PieChart width={200} height={200}>
        <Pie
          data={data}
          cx="50%"
          cy="50%"
          innerRadius="60%"
          outerRadius="90%"
          fill="#8884d8"
          paddingAngle={0}
          dataKey="value"
        >
          {data.map((entry, index) => (
            <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
          ))}
        </Pie>
      </PieChart>
      <div
        style={{
          fontSize: "32px",
          fontWeight: "bold",
          position: "absolute",
          color: checkStatusState(componentData),
          top: "30%",
          left: "50%",
          transform: "translate(-50%, -50%)",
        }}
      >
        {componentData}%
      </div>
      <div className="">
        <div className="text-center">30 Days Backup Success Rate</div>
        <div className="text-center font-bold" style={{ color: statusState.color }}>
          {statusState.text}
        </div>
      </div>
    </div>
  );
}`

File: front/src/page/AddAudit.jsx [hunk 2]
```diff
@@ -145,6 +168,238 @@ export const AddAudit = () => {
...
+                Online On Premise Servers
+              </label>
+              <select
+                required
+                id="online-premise-servers-status"
+                name="online-premise-servers-status"
+                className=" w-[30%] bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
+              >
+                <option value="Protected">Protected</option>
+              </select>
```
Comment (konantovich) [inline front/src/page/AddAudit.jsx:196]: We can create and use our own select to remove code duplication


const CustomSelect = ({children, id, name}) => {
  return <select
  required
  id={id}
  name={name}
  className="mb-5 w-[50%] bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
>{children} </select>}


Use example:
`   <CustomSelect
                id="overall-information"
                name="overall-information"
              >
                <option value="Bronze">Bronze</option>
                <option value="Silver">Silver</option>
                <option value="Gold">Gold</option>
                <option value="Platinum">Platinum</option>
                <option value="No contract">No contract</option>
              </CustomSelect>`

File: front/src/page/AddAudit.jsx [hunk 3]
```diff
@@ -145,6 +168,238 @@ export const AddAudit = () => {
...
+                class="block text-sm font-medium text-gray-900 dark:text-white"
+              >
+                Test File Recovery
+              </label>
+              <input
+                id="file-test-recovery"
+                name="file-test-recovery"
+                type="date"
+                className="w-[50%] bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
+              />
```
Comment (konantovich) [inline front/src/page/AddAudit.jsx:276]: And we can create custom input too

File: front/src/components/audit/PieChartWithPaddingAngle.jsx (hunk 1 above)
Comment (microna) [reply front/src/components/audit/PieChartWithPaddingAngle.jsx]: fixed

File: front/src/page/AddAudit.jsx (hunk 2 above)
Comment (microna) [reply front/src/page/AddAudit.jsx:196]: done

File: front/src/page/AddAudit.jsx (hunk 3 above)
Comment (microna) [reply front/src/page/AddAudit.jsx:276]: agreed

**Review by konantovich — APPROVED** [review]

**Pull request merged**